### PR TITLE
Issue 1246: Scripting the whole release procedure

### PR DESCRIPTION
**Pull request by sijie**
Descriptions of the changes in this PR:

*Motivation*

Releasing bookkeeper becomes tricky after introducing `circe-checksum`, since we need to ensure linux jni library is shipped as part of `circe-checksum`.

*Solution*

- switch to use docker for generating bookkeeper release
- scripting the whole release procedure as many as possible

Master Issue: #1246 

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

### CODE REVIEW

**Comment by sijie** [comment]
now this is ready for review

File: dev/release/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM maven:3.5.0-jdk-8
```
Comment (eolivelli) [inline dev/release/Dockerfile]: Maybe it is better to latest maven 3.5.3

**Comment by sijie** [comment]
> It is not clear to me how gpg key is passed to the container, can you give me a pointer please?

We don’t pass gpg key. The docker add same userid and homedir, so the gpg agent will load them when docker image is launched

**Comment by sijie** [comment]
The changes are docs changes only. Travis CI passed. The java 9 CI failure is unrelated.

**Comment by sijie** [comment]
IGNORE CI

**Pull request closed without merging**